File: class9/ex46_redoTheQuestions/class5/ex20_sum.c
/*Desenvolva o código em C de um programa que soma números
inteiros informados pelo usuário até que um número negativo seja
informado e então imprime o resultado da soma na tela.*/

#include <stdio.h>

int main() {

    int n;
    int sum = 0;

    printf("Digite um número: ");
    scanf("%d", &n);

    while (n >= 0) {

        sum += n;

        printf("Digite um número: ");
        scanf("%d", &n);

        if (n < 0) {

            sum += n;
            printf("O resultado da soma de é %d\n", sum);
        }
    }

    return 0;
}

File: GO/class5/ex20_sum.go
/*Desenvolva o código em C de um programa que soma números
inteiros informados pelo usuário até que um número negativo seja
informado e então imprime o resultado da soma na tela.*/

package main

import "fmt"

func main() {

	var n int
	sum := 0

	fmt.Print("Digite um número: ")
	fmt.Scanln(&n)

	for n >= 0 {

		sum += n

		fmt.Print("Digite um número: ")
		fmt.Scanln(&n)

		if n < 0 {

			sum += n
			fmt.Printf("O resultado da soma é %d\n", sum)
		}
	}
}
